Create Security Team Folder and place holder files for items that need to be added.

akgraner opened the pull request.

kimwnasptd commented:
Thanks for sending the PR @akgraner! Took a small look and had some nits regarding the namings and folder structure.

By being inspired at what we did for the release team https://github.com/kubeflow/community/tree/master/releases what I'd suggest would be:
```
security/
  meetings/
    01-02-2023.md
    01-03-2023.md
    ...
  policies-and-procedures.md
  roadmap.md
  security-team.md
```

The above are minor nits, but I believe will help ensure we use similar names for similar files across the project.

akgraner commented:
@kimwnasptd thank you! I'll get those fixed. Appreciate the feedback.

akgraner commented:
delete the original directory and folders and added the directories and folders in the correct format

kimwnasptd commented:
Thanks!

/lgtm

eslerm reviewed (approved):
LGTM +1

Perhaps add a line to `roadmap.md` and `security-team.md` that describes what they are for.

juliusvonkohout commented on security/meetings/01-18-2023.md:
```diff
@@ -0,0 +1,79 @@
+
+* **(_Done_)Will we need Google to add a repo?** 
+
+    * Amber to raise an issue and ask that Google create a Kubeflow/security repo with Julis, Amber and Alex as owners and we will add others as we go forward with this team - Need to raise an issue within kubeflow/kubeflow or kubeflow/transition to get this repo added.  Will email James, Thea and Josh to see where to raise the issue. **UPDATE:** We now have a security directory under kubeflow/ku ...
```
I think it should be Julius, not Julis

juliusvonkohout replied:
I think the owners file is missing.

james-jwu commented on security/security-team.md:
```diff
@@ -0,0 +1 @@
+
```
Empty file?

james-jwu commented on security/README:.md:
```diff
@@ -0,0 +1,12 @@
+# Kubeflow Security Team
```
Do you want to add a OWNERS file to give permission to additional folks?

juliusvonkohout commented:
i recommend to upload the presentation as well 
[Hardening Kubeflow Security for Enterprise Environments - Julius von Kohout, DPDHL & Diana Dimitrova Atanasova, VMware.pdf](https://github.com/kubeflow/kubeflow/files/11449689/Hardening.Kubeflow.Security.for.Enterprise.Environments.-.Julius.von.Kohout.DPDHL.Diana.Dimitrova.Atanasova.VMware.pdf)

kimwnasptd commented on security/README:.md:
```diff
@@ -0,0 +1,12 @@
+* Owners and Team Members
+* Roadmaps
+
+## How to participate in the Kubeflow Security Team
```
Let's add links for 
1. The Calendar https://calendar.google.com/calendar/embed?src=kubeflow.org_7l5vnbn8suj2se10sen81d9428%40group.calendar.google.com
2. The drive with the recordings (link?)

kimwnasptd reviewed (commented):
Did one more pass. 

Should we remove the `meetings` folder? We have a [gdoc](https://docs.google.com/document/d/1xGkg9GuO2OjvYhdONJFbSrpF66UKhtYonczttJoTv3s/edit#heading=h.fmin4de6g5bt) for the meeting notes so it would just add extra maintenance burden to copy the notes from the doc to GH every time. 

Instead I'd propose to update the `README.md` to point to all necessary information, like the gdoc, the calendar, drive etc

akgraner commented:
Thanks everyone for all the feedback. Much appreciated. 
Here's what's been done. 

* Fixed all the empty folders
* Added the first iteration of the Security Team Policies and Procedures 
* Added the OWNERS file
* Deleted the Team Members folder and the Meetings folders
* Updated the meeting link 
* Updated the Notes link

@kimwnasptd and @james-jwu can you please review.  I think it should be good to go now.

zijianjoy commented:
/assign @theadactyl @james-jwu

zijianjoy commented on security/policies-and-procedures.md:
```diff
@@ -0,0 +1,24 @@
+
+The Kubeflow Security team (listed below as “we”) will take a multi-level, transparent approach to the security of Kubeflow (product/code). At this time we will base this effort to align with each release.  As noted above as we get more volunteers to help with these efforts we will continue to iterate on the cadence.  
+
+1. We will encourage Kubeflow users to open issues with the respective WG that the vulnerability was found and tag it with the “security” tag. If they don’t know how to do that or don’t feel comfortable doing so, they can also contact us via our #security slack channel.  And if a private mailing list/group is needed we will set that up. 
```
It is strongly recommended that we don't add a security tag to an open issue. 

You can find a security template put up by CNCF in https://github.com/cncf/tag-security/tree/main/project-resources.

Reference: https://github.com/cncf/tag-security/blob/main/project-resources/templates/SECURITY.md#reporting-a-vulnerability

Most CNCF projects right now use email to notify security issue. GitHub just released their private vuln reporting feature (https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability). Either email shared among the people who will be responsible for vuln response or this new GitHub reporting channel are fine.

akgraner replied:
@zijianjoy, I agree, but when I asked about getting an email created for the security team there was some push back. Since Google has to create the email address can you all create a security@kubeflow.org email for the team. The Working Groups asked that security issues related respective working groups be filed their, so (at the time) without being able to create an official email for the group and the resources we had available we came up with the above process. I'll take a look at the resources you have listed and rework the process and bring it up to the group on the 24th at our next meeting. 

In order to get this PR approved if I change this file to say we are currently working on the process and a link to our meeting notes would this work for sake of approving this?

zijianjoy replied:
We are in discussion about creating such email address. Make sense to track this effort in another PR. At the meantime, James Wu has LGTMed your PR here.

akgraner commented:
@james-jwu and @theadactyl  - in addition to the list from earlier this morning, I also removed the wording for the policy and procedures and will rework this based on @zijianjoy's suggestions above.  I removed @jbottum from owner's list.  Can this PR now be approved?

james-jwu commented on security/policies-and-procedures.md:
```diff
@@ -0,0 +1,12 @@
+
+Kubeflow Security Team [Policy and Procedure Working Document] (https://docs.google.com/document/d/1vw_efQyYG_zWEoL-vk9mZQX5p7fOcl3RTG6pcxMhYKI/edit?usp=sharing) - Please note this is only a **WORKING** Document. 
+
+We are always looking for Kubeflow community users and contributoes to help us. If you would like to help the security team with these docs or other items please consider doing the following:
```
Typo: contributors

akgraner replied:
@james-jwu - Thank you! Fixed this typo.

james-jwu commented:
/lgtm

akgraner commented on security/policies-and-procedures.md:
```diff
@@ -0,0 +1,12 @@
+We are always looking for Kubeflow community users and contributoes to help us. If you would like to help the security team with these docs or other items please consider doing the following:
+* Join the Kubeflow Slack Security Channel: #security
+* Join our meetings bi-weekly on the Wednesday's at 8am Pacific/10am Central. (To find out what that is in your time zone go to [timeanddate.com](https://www.timeanddate.com/). This meeting is on the Kubeflow community calendar; however if you would like to be added to the invite please reach out to Amber Graner on Slack (@akgraner) The invite information is also pinned to the security slack chan ...
+* Meeting Notes: If you would like to see what we have previously discusses you can see the meetings folder in the security folder or you can go to our [gdoc](https://docs.google.com/document/d/1xGkg9GuO2OjvYhdONJFbSrpF66UKhtYonczttJoTv3s/edit?usp=sharing)
```
@james-jwu  - Thank you! Fixed this typo.

akgraner commented:
@james-jwu fixed the typos and the broken hyperlink which removed your lgtm. Sorry about that. Can you re-approve? Please and thank you.

james-jwu commented:
/lgtm
/approve
